From 104326ff2424cc831790a522f891fbdc806987c3 Mon Sep 17 00:00:00 2001
From: Manish Patel <manish.patel@xzadium.com>
Date: Wed, 22 Oct 2014 16:48:39 +0530
Subject: [PATCH 5/5] Make HDMI independent from u-boot environment

Signed-off-by: Manish Patel <manish.patel@xzadium.com>
---
 drivers/video/wmt/vout-wmt.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/video/wmt/vout-wmt.c b/drivers/video/wmt/vout-wmt.c
index a82a612..323f927 100644
--- a/drivers/video/wmt/vout-wmt.c
+++ b/drivers/video/wmt/vout-wmt.c
@@ -950,6 +950,7 @@ const char *vout_sys_parm_str[] = {"wmt.display.param", "wmt.display.param2"};
 int vout_check_display_info(struct vout_init_parm_t *init_parm)
 {
 	char buf[100];
+	const char *display_param[] = { "2:0:24:1024:600:60", "4:0:24:1024:600:60" };
 	int varlen = 100;
 	unsigned int parm[10];
 	vout_t *vo = 0;
@@ -961,9 +962,7 @@ int vout_check_display_info(struct vout_init_parm_t *init_parm)
 
 	/* [uboot parameter] display param : type:op1:op2:resx:resy:fps */
 	for (i = 0; i < 2; i++) {
-		if (wmt_getsyspara((char *)vout_sys_parm_str[i], buf, &varlen))
-			continue;
-
+		strncpy(buf, display_param[i], 19);
 		DBG_MSG("display param %d : %s\n", i, buf);
 		vpp_parse_param(buf, (unsigned int *)parm, 6, 0);
 		MSG("boot parm vo %d opt %d,%d, %dx%d@%d\n", parm[0],
-- 
1.8.3.2

